core/src/main/scala/cats/TraverseFilter.scala

```diff
@@ -122,6 +122,9 @@ trait TraverseFilter[F[_]] extends FunctorFilter[F] {
   override def mapFilter[A, B](fa: F[A])(f: A => Option[B]): F[B] =
     traverseFilter[Id, A, B](fa)(f)
 
+  def mapAccumulateFilter[S, A, B](init: S, fa: F[A])(f: (S, A) => (S, Option[B])): (S, F[B]) =
```

Comment:
I feel that a scaladoc comment (perhaps, with a short usage example) wouldn't hurt and could come handy here.

Reply:
Looks good to me, thanks!